lib/vtls/openssl.c

```diff
@@ -1878,15 +1878,42 @@ static void ossl_close(struct Curl_cfilter *cf, struct Curl_easy *data)
 
   if(backend->handle) {
     if(cf->next && cf->next->connected) {
-      char buf[32];
-      /* Maybe the server has already sent a close notify alert.
-         Read it to avoid an RST on the TCP connection. */
-      (void)SSL_read(backend->handle, buf, (int)sizeof(buf));
-
-      (void)SSL_shutdown(backend->handle);
+      char buf[1024];
+      int nread, err;
+      long sslerr;
 
       ERR_clear_error();
+      if(SSL_shutdown(backend->handle)) {
+        CURL_TRC_CF(data, cf, "SSL shutdown finished");
+      }
+      else {
+        nread = SSL_read(backend->handle, buf, (int)sizeof(buf));
```

Comment:
I also wonder how this works for FTPS TLS 1.3 session tickets on the data connection as reported in #6149. There was a report that @mkauf's change to SSL_read first worked to read any unread tickets in the queue, which otherwise would not be read if data channel is upload only.

Reply:
@jay I assume you mean something like the change I added just now?